server/trafficrules/src/main/java/com/coffeejug/trafficrules/service/UserService.java

```diff
@@ -0,0 +1,61 @@
+package com.coffeejug.trafficrules.service;
+
+import com.coffeejug.trafficrules.db.Progress;
+import com.coffeejug.trafficrules.db.User;
+import com.coffeejug.trafficrules.dto.UserCodeDto;
+import com.coffeejug.trafficrules.projection.ProgressPrj;
+import com.coffeejug.trafficrules.repository.UserRepository;
+import com.coffeejug.trafficrules.util.CodeGenerator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class UserService {

```

Comment:
I strongly recommend having a unit and international tests for our business logic

Reply:
I can't understant crearly what does it mean.

Reply:
We need to cover the code with tests :)

Reply:
I haven't used JUnit before :-(

Reply:
That's a perfect opportunity to learn it, isn't it?